src/pipx/util.py

```diff
@@ -75,14 +75,9 @@ def get_venv_paths(root: Path) -> Tuple[Path, Path]:
         return bin_path, python_path
 
 
-def get_script_output(interpreter: Path, script: str, *args) -> str:
-    # Make sure that Python writes output in UTF-8
-    env = os.environ.copy()
-    env["PYTHONIOENCODING"] = "utf-8"
```

Comment:
This should be applied in the `run_subprocess` function

Reply:
if `subprocess.run()` is called with the `universal_newlines` option set to true (as it is in `run_subprocess()`) then stdout and stderr are already decoded text.  I think the PYTHONIOENCODING stuff is only needed to decode bytes, but with the `universal_newlines` this is not necessary.

https://docs.python.org/3.6/library/subprocess.html#subprocess.run
> If encoding or errors are specified, or universal_newlines is true, file objects for stdin, stdout and stderr are opened in text mode using the specified encoding and errors or the io.TextIOWrapper default. Otherwise, file objects are opened in binary mode.

Reply:
> or the io.TextIOWrapper default

If the encoding isn't specified, then the default is `locale.getpreferredencoding` (https://docs.python.org/3/library/locale.html#locale.getpreferredencoding)

There are various "fun" nuances with this:

> Return the encoding used for text data, according to user preferences. User preferences are expressed differently on different systems, and might not be available programmatically on some systems, so this function only returns a guess.
On some systems, it is necessary to invoke setlocale() to obtain the user preferences, so this function is not thread-safe. If invoking setlocale is not necessary or desired, do_setlocale should be set to False.
On Android or in the UTF-8 mode (-X utf8 option), always return 'UTF-8', the locale and the do_setlocale argument are ignored.
Changed in version 3.7: The function now always returns UTF-8 on Android or if the UTF-8 mode is enabled.

The `PYTHONIOENCODING` env variable is described like this:

> If this is set before running the interpreter, it overrides the encoding used for stdin/stdout/stderr, in the syntax encodingname:errorhandler. Both the encodingname and the :errorhandler parts are optional and have the same meaning as in str.encode().

I think using it defensively to prevent any edge cases from popping up makes sense.

Reply:
OK.  I wasn't worried about the performance hit--my thought was that getting the text decoded by a standard library instead of us doing it by hand might be simpler and more reliable.  And I was trying to disturb the environment variables less.

But I can add the PYTHONIOENCODING back.

Reply:
We are letting it be decoded by the standard library since `universal_newlines` is true. But we are making extra sure the bytes are decoded to text the right format by specifying the format rather than leaving it up to the system's configuration.

Reply:
We should also specify the encoding on the `subprocess.run` call to make sure it's decoded properly.

Reply:
I wish the `subprocess` docs were clearer.  My assumption was that subprocess would use the encoding that is being used to make the bytes in the first place, but I guess one can't tell from the docs.

Yes, I was alreading setting the `encoding` to "utf-8" in `subprocess.run()` :+1:

Reply:
> I wish the subprocess docs were clearer. My assumption was that subprocess would decode using the encoding that is being used to make the bytes in the first place, but I guess one can't tell from the docs.

Totally agree